lib/src/rules/cascade_invocations.dart

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.cascade_invocations;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/linter.dart';
+
+const _desc = r'Cascade consecutive method invocations on the same reference.';
+
+const _details = r'''
+
+**DO** Use the cascading style when succesively invoking methods on the same
+ reference.
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**BAD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference.firstMethod();
+someReference.secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass()
+    ..firstMethod()
+    ..secondMethod();
+```
+
+**GOOD:**
+```
+SomeClass someReference = new SomeClass();
+...
+someReference
+    ..firstMethod()
+    ..secondMethod();
+```
+
+''';
+
+class CascadeInvocations extends LintRule {
+  _Visitor _visitor;
+  CascadeInvocations()
+      : super(
+            name: 'cascade_invocations',
+            description: _desc,
+            details: _details,
+            group: Group.style) {
+    _visitor = new _Visitor(this);
+  }
+
+  @override
+  AstVisitor getVisitor() => _visitor;
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+  _Visitor(this.rule);
+
+  @override
+  void visitMethodInvocation(MethodInvocation node) {
```

Comment:
Consider visiting ExpressionStatement rather than MethodInvocation. That way you won't have to test the parent of every method invocation nested within other expressions.